### PR TITLE
Added browser parameter in package.json pointing to transpiled dist

### PR DESCRIPTION
**Pull request by aydrian**
Added rollup.js to transpile and bundle the library.
Added the `browser` parameter in `package.json` and pointed it to the transpiled dist.
Added a `prepublishOnly` script to ensure the transpiled version is built.

### CODE REVIEW

File: rollup.config.js [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+          'env',
+          {
+            modules: false
+          }
+        ]
+      ]
+    })
+  ],
+  external: ['bson'],
+}
```
Comment (daprahamian) [inline rollup.config.js]: Add a newline

File: .gitignore [hunk 2]
```diff
@@ -35,3 +35,5 @@ jspm_packages
 
 # Optional REPL history
 .node_repl_history
+
+dist
```
Comment (daprahamian) [inline .gitignore]: Add a newline

File: rollup.config.js (hunk 1 above)
Comment (aydrian) [reply rollup.config.js]: Done

File: .gitignore (hunk 2 above)
Comment (aydrian) [reply .gitignore]: Done

**Comment by hswolff** [comment]
Did rollup for yah @daprahamian

**Pull request merged**